application/src/main/java/org/thingsboard/server/service/security/auth/oauth2/BasicMapperUtils.java

```diff
@@ -69,7 +73,12 @@ public static String getTenantName(String email, Map<String, Object> attributes,
     public static String getStringAttributeByKey(Map<String, Object> attributes, String key) {
         String result = null;
         try {
-            result = (String) attributes.get(key);
+            if (key.startsWith("$")) {
+                Object json = Configuration.defaultConfiguration().jsonProvider().parse(JacksonUtil.toString(attributes));
```

Comment:
It is not a good idea to parse attributes every time we need to extract something. It is better to create a new method that will accept JSON object and key. Also, what if JsonPath.read will fail? There might be a key that starts with $ but is not a JSON expression, right?